airflow/config_templates/config.yml

```diff
@@ -2053,3 +2053,13 @@
       type: string
       example: ~
       default: "NamedHivePartitionSensor"
+- name: ssh
+  description: ~
+  options:
+    - name: private_keys_algorithm_support
+      description: |
+        Comma separated list of ssh algorithms names (following `ssh-keygen` naming) supported by
+        SSH hook
+      type: string
+      example: ~
+      default: "dsa,ecdsa,ed25519,rsa"
```

Comment:
This config value doesn't appear to be used anywhere.

Either we should remove it, or use it. Given this needs a source-code change to add a new supported type, is there any value in having this config section?

Reply:
@NBardelot Did you have a use case in mind where you would like to configure this value -- given this is just to support the SSH key _given by the user_ do we really need this configurable.

Reply:
Where I work, Airflow is given to teams of users as a platform:

  * my team builds the infrastructure and provides Airflow with a standard configuration
  * the teams use their instance of Airflow

Our standard configuration includes the decisions about security. We need to be able to enforce the security team choice of algorithms used.

Reply:
I don't think I have a strong enough argument for either option on this. Does Airflow configuration typically allow for limitations on user input like this?

Let me know what you guys decide and I'll change it in either direction (by either removing the config altogether, or adding reconciliation logic that will remove algos not specified in the config)

Reply:
@ashb what do you think?

Reply:
Is that really enforce anything? Any one can create a custom SshHook and overwrite this security measure. Isn't it?

Reply:
Yeah, this doesn't really feel like a config setting adds anything here -- this should be enforced on the connecting server side rather than here.

@brighton1101 Lets remove this part please.

(Sorry for letting this languish for sooo long)

Reply:
Hi, I've not been very available lately but I'd like to say that putting the only control on the server-side is not the intended way the protocols like SSH behave, in my opinion. Protocol negociation is made to reflect both the client and server interests. 

An Airflow instance might very well connect to a server that is not under the control of its administrator. It is typically the case in large companies. The administrator should be able to implement a policy, limiting protocols, without depending on Paramiko or the server administrator.

Reply:
@ashb I'll try to get this done by end of day tomorrow 👍